src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2791,6 +2792,67 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
             return objectCreation;
         }
 
+        public override IOperation VisitDeconstructionAssignment(IDeconstructionAssignmentOperation operation, int? captureIdForResult)
+        {
+            // If the assignment target is a tuple, we want to decompose the tuple and push each element back onto the stack, so that if the value
+            // has control flow the individual elements are captured. Then we can recompose the tuple after operation.Value has been visited.
+            // We do this to keep the graph sane, so that users don't have to track a tuple captured via flow control when it's not really
+            // the tuple that's been captured, it's the operands to the tuple.
+            pushTargetAndUnwrapTupleIfNecessary(operation.Target);
+            IOperation visitedValue = Visit(operation.Value);
+            IOperation visitedTarget = popTargetAndWrapTupleIfNecessary(operation.Target);
+
+            return new DeconstructionAssignmentExpression(visitedTarget, visitedValue, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+
+            // Recursively push nexted values onto the stack for visiting
+            void pushTargetAndUnwrapTupleIfNecessary(IOperation value)
+            {
+                if (value.Kind == OperationKind.Tuple)
+                {
+                    var tuple = (ITupleOperation)value;
+
+                    foreach (IOperation element in tuple.Elements)
+                    {
+                        pushTargetAndUnwrapTupleIfNecessary(element);
+                    }
+                }
+                else
+                {
+                    _evalStack.Push(Visit(value));
+                }
+            }
+
+            // Recursively pop nested tuple values off the stack after visiting
+            IOperation popTargetAndWrapTupleIfNecessary(IOperation value)
+            {
+                if (value.Kind == OperationKind.Tuple)
+                {
+                    var tuple = (ITupleOperation)value;
+                    var elementBuilder = ArrayBuilder<IOperation>.GetInstance();
```

Comment:
>GetInstance() [](start = 66, length = 13)

Do we know the size of the builder that we need?

Reply:
We do, I'll update.

---
In reply to: [181503256](https://github.com/dotnet/roslyn/pull/26108#discussion_r181503256) [](ancestors = 181503256)